packages/worker/src/api/controllers/global/users.ts

```diff
@@ -185,9 +186,23 @@ export const getAppUsers = async (ctx: Ctx<SearchUsersRequest>) => {
 export const search = async (ctx: Ctx<SearchUsersRequest>) => {
   const body = ctx.request.body
 
-  // TODO: for now only one supported search key, string.email
-  if (body?.query && !userSdk.core.isSupportedUserSearch(body.query)) {
-    ctx.throw(501, "Can only search by string.email or equal._id")
+  // TODO: for now only two supported search keys; string.email and equal._id
+  if (body?.query) {
+    // Clean numeric prefixing. This will overwrite duplicate search fields,
+    // but this is fine because we only support a single custom search on
+    // email and id
+    for (let filters of Object.values(body.query)) {
+      if (filters && typeof filters === "object") {
+        for (let [field, value] of Object.entries(filters)) {
+          delete filters[field]
+          filters[db.removeKeyNumbering(field)] = value
```

Comment:
Is there a risk of this defeating the purpose of the numeric keying, e.g. if we have sent up `999:email` and `1:email`, this will flatten them in an order depending on how the object gets enumerated (likely deterministic but possibly confusing) we could probably write a quick test case and adjust the behaviour in someway to handle it.

Reply:
It will yea, but I explained in the comment that the underlying search mechanism here only supports a single filter anyway, so flattening is fine since we would never have used both in the first place.

Reply:
Should we add a case to prefer the `999:email` over all others? As this is one set by the system this is really the one we want to use.

Reply:
Hmm the `999` is the one sent by relationship pickers (large just to avoid clashing with other filters), but if you just did a normal filter yourself on the users table then you could easily sent up 2 email filters with a `1` and `2` prefix for example, and then you still wouldn't know which would win. I'm leaning towards just erroring for correctness here I think, since that's the only way to avoid us having to be opinionated in some hidden way.

Reply:
yeah I think that works, an error makes it easier to understand what is happening if we have to debug it at least!